cmd/dockerd/daemon_windows.go

```diff
@@ -89,9 +89,18 @@ func newCgroupParent(config *config.Config) string {
 	return ""
 }
 
-func (cli *DaemonCli) initContainerd(_ context.Context) (func(time.Duration) error, error) {
-	system.InitContainerdRuntime(cli.ContainerdAddr)
-	return nil, nil
+func (cli *DaemonCli) initContainerd(ctx context.Context) (func(time.Duration) error, error) {
+	if cli.ContainerdAddr != "" {
+		// use system containerd at the given address.
+		defer func() { system.EnableContainerdRuntime(cli.ContainerdAddr) }()
```

Comment:
This defer needs to be outside the if block, or it won't be applied to the code-path that calls `cli.initializeContainerd(ctx)`.